run-basic-batch-operators-test.sh

```diff
@@ -232,6 +232,44 @@ wait_package_ok() {
 	done
 }
 
+wait_all_packages_ok() {
+	local \
+		start_time \
+		timeout_seconds=600
+
+	prev_count=$(get_packeges)
+	start_time=$(date +%s 2>&1) || {
+		echo "date failed with error $?: $start_time" >>"$LOG_FILE_PATH"
+		return 0
+	}
+
+	# wait until package number is stable
+	while true; do
+		curr_count=$(get_packeges)
+		if [ "${curr_count}" -ne "${prev_count}" ] || [ "${curr_count}" -eq 0 ]; then
+			prev_count=${curr_count}
+		else
+			return 0
+		fi
+
+		curr_time=$(date +%s)
+		elapsed_time=$((curr_time - start_time))
+		# If elapsed time is greater than the timeout report failure
+		if [ "$elapsed_time" -ge "$timeout_seconds" ]; then
+			echo_color "$RED" "Timeout reached $timeout_seconds seconds waiting for packagemanifests to be reachable."
+			return 1
+		fi
+
+		# Otherwise wait a bit
+		echo_color "$BLUE" "Waiting for packages to be reachable..."
+		sleep 5
+	done
+}
+
+get_packeges() {
+	oc get packagemanifest -n ${OPERATOR_CATALOG_NAMESPACE} -o json | jq -r '.items[] | select(.status.catalogSource == "'${OPERATOR_CATALOG_NAME}'") | .metadata.name' | wc -w
```

Comment:
Add error handling, e.g.:
command || {
  local err=$?
  echo >&2 "Command failed with $err."
  exit $err
}

Reply:
we want to keep on checking the packages, if there will be an error the elapsed time will reach the limit and exit (line 258)